### ISSUE TITLE
Google sign-in without firebase

### ISSUE THREAD

**Issue by omarbelkhodja** [flutter/flutter#36673]
In most of tutorials on the web, I can see that the plugin google_sign_in doesn't require firebase. In pub page, it is mentioned that the application must be registered in the following link : 
https://developers.google.com/mobile/add?platform=android

That link is now redirecting to firebase's homepage. So my question is: 
Is it still possible to Sign-in without firebase ? Where should I register the app in that case ?

**Comment by thehitechpanky** [comment]
Is this planned any sooner?

**Comment by truthmast** [comment]
For using `google_sign_in` with a plain Google Cloud project, here's what I'm doing:

- From the GCP Console for your project, go to __APIs & Services__ > __Credentials__ (https://console.cloud.google.com/apis/credentials)
- Click the __Create Credentials__ button, select __OAuth Client ID__
- On the next screen, select __Android__ as the Application Type
- Fill in __Signing-certificate fingerprint__
- Fill in __Package name__

It seems like creating a Firebase app takes care of this step for you (and does a bunch of other things to your GCP project as well), but it's the only thing required for basic login as far as I can tell.

**Comment by thehitechpanky** [comment]
This method does not give idtoken

**Comment by nvma** [comment]
It does, you need to make sure to register both SHA-1 and SHA-256 keys in project settings.

**Comment by BondarenkoStas** [comment]
@omarbelkhodja Is the issue still relevant after comments above?

**Comment by anoop-ananthan** [comment]
Yes, this is a big problem. There don't seem to be any documentation except https://github.com/flutter/flutter/issues/36673#issuecomment-520446732. Thanks @truthmast

**Comment by christophemacabiau** [comment]
Has anyone a solution to this issue?

**Comment by TahaTesser** [comment]
From https://github.com/flutter/flutter/issues/73455 by @avioli
https://github.com/flutter/flutter/issues/73455#issuecomment-759116031 should be documented to avoid running into issues

**Comment by aacassandra** [comment]
works fine for android and not for ios. i spend a lot of time on ios regarding google login without firebase

**Comment by tiagocdr** [comment]
Any Solutions for IOS ? Every solution I have found has been only for android

**Comment by toan-ngvan** [comment]
I find the solution. Just download the `.plist` file from gg cloud, rename the file to `GoogleService-Info.plist` and put it in Runner.

**Comment by CodeCakePie** [comment]
Any updates on this?

**Comment by zzundalek** [comment]
Please at least clarify it:
- Is firebase **required** for google_sign_in to work on iOS?
-  If yes add it to plugin description if no add working example without firebase

Thanks a lot.

**Comment by mhrst** [comment]
_Updating my instructions from a previous comment to include steps for setting up iOS_

For using `google_sign_in` with a plain Google Cloud project (no Firebase required), here's what I'm doing:

### Android
* From the GCP Console for your project, go to **APIs & Services** > **Credentials** (https://console.cloud.google.com/apis/credentials)
* Click the **Create Credentials** button, select **OAuth Client ID**
* On the next screen, select **Android** as the Application Type
* Fill in **Signing-certificate fingerprint**
  * For instructions on finding your signing-certificate fingerprint, https://developers.google.com/android/guides/client-auth)
* Fill in **Package name**

### iOS
* From the GCP Console for your project, go to **APIs & Services** > **Credentials** (https://console.cloud.google.com/apis/credentials)
* Click the **Create Credentials** button, select **OAuth Client ID**
* On the next screen, select **iOS** as the Application Type
* Fill in **Bundle ID**
* After clicking **Create** a dialog will open with a **DOWNLOAD PLIST** button -- Grab this file and rename it to `GoogleService-Info.plist`
* Open Xcode and put this plist file into the `Runner` folder

**Comment by anqit** [comment]
Everything online says `google_sign_in` can be used on android without having to explicitly set the `client_id` when instantiating the `GoogleSignIn` instance. When I was using firebase, this worked fine. Now, I'm trying without firebase and am not able to get an id_token unless not only I do specify the `client_id` but also make sure to not use the android one I created but use the web one instead. Is this because I have misconfigured something or is this expected behavior? I definitely must echo the sentiment that documentation for this basic functionality seems to be sorely lacking.

**Comment by wildsylvan** [comment]
same as @anqit

[here ](https://github.com/flutter/flutter/issues/20903) are some workarounds

**Comment by jaybe78** [comment]
I followed this documentation https://medium.com/codebrew/flutter-google-sign-in-without-firebase-3680713966fb
Since I've changed  classpath 'com.google.gms:google-services:4.3.15'
I'm getting the following issue 
requires google-services.json file where that is coming from firebase
this is confusing ??

**Comment by nevsnirG** [comment]
> _Updating my instructions from a previous comment to include steps for setting up iOS_
> 
> For using `google_sign_in` with a plain Google Cloud project (no Firebase required), here's what I'm doing:
> ### Android
> 
>     * From the GCP Console for your project, go to **APIs & Services** > **Credentials** (https://console.cloud.google.com/apis/credentials)
> 
>     * Click the **Create Credentials** button, select **OAuth Client ID**
> 
>     * On the next screen, select **Android** as the Application Type
> 
>     * Fill in **Signing-certificate fingerprint**
>       
>       * For instructions on finding your signing-certificate fingerprint, https://developers.google.com/android/guides/client-auth)
> 
>     * Fill in **Package name**
> 
> 
> ### iOS
> 
>     * From the GCP Console for your project, go to **APIs & Services** > **Credentials** (https://console.cloud.google.com/apis/credentials)
> 
>     * Click the **Create Credentials** button, select **OAuth Client ID**
> 
>     * On the next screen, select **iOS** as the Application Type
> 
>     * Fill in **Bundle ID**
> 
>     * After clicking **Create** a dialog will open with a **DOWNLOAD PLIST** button -- Grab this file and rename it to `GoogleService-Info.plist`
> 
>     * Open Xcode and put this plist file into the `Runner` folder

Sadly I still get this error:
[_handleGoogleLogin] Google login failed: Exception: Failed to login with Google: PlatformException(sign_in_failed, com.google.android.gms.common.api.ApiException: 10: , null, null)
                     Exception: Failed to login with Google: PlatformException(sign_in_failed, com.google.android.gms.common.api.ApiException: 10: , ...

I verified the SHA-1 hash a million times. Generated a new keystore and tried that. Package name matches in the build.gradle and AndroidManifest too. Tried making new Android credentials in the Google Console. Still nothing unfortunately.

**Comment by Dzivo** [comment]
Having the same problem even with firebase i have flavors setup so something can be an issue there. Really anoying errors dont say whats wrong. Shouldnt this be fixed asap ? This is like breaking of the breakings bug
